Update Linux examples to new runner directory

robert-ancell opened the pull request:
This was changed in https://github.com/flutter/flutter/pull/153812

robert-ancell committed:
Update Linux examples to new runner directory

stuartmorgan commented:
test-exempt: code refactor with no semantic change

stuartmorgan-g reviewed (approved):
LGTM

robert-ancell committed:
Merge branch 'main' into linux-examples-runner-dir

chunhtai commented:
not sure why ci failed, @robert-ancell can you rebase and see if it passes the ci

robert-ancell commented:
It seems to be related to running the API test and the runner directory not existing for some reason. Still investigating.

robert-ancell committed:
Merge branch 'main' into linux-examples-runner-dir

robert-ancell committed:
Fix smoke test failing with new runner directory

Instead of calling "flutter test" in examples/api/ call it from examples/api/integration_test

robert-ancell commented:
I've got the build to complete by changing the working directory for the smoke tests.

Without this change the build fails when you run:
```
$ cd dev/tools
$ dart examples_smoke_test.dart
00:00 +0: loading /home/bob/git/flutter/flutter/examples/api/integration_test/smoke_integration_test.dart
Building Linux application...                                   
CMake Error at CMakeLists.txt:58 (add_subdirectory):
  add_subdirectory given source "runner" which is not an existing directory.


CMake Error at CMakeLists.txt:61 (add_dependencies):
  Cannot add target-level dependencies to non-existent target
  "flutter_api_samples".

  The add_dependencies works for top-level logical targets created by the
  add_executable, add_library, or add_custom_target commands.  If you want to
  add file-level dependencies see the DEPENDS option of the add_custom_target
  and add_custom_command commands.


CMake Error at CMakeLists.txt:67 (set_target_properties):
  set_target_properties Can not find target to add properties to:
  flutter_api_samples


CMake Error at CMakeLists.txt:94 (install):
  install TARGETS given target "flutter_api_samples" which does not exist.


00:00 +0 -1: loading /home/bob/git/flutter/flutter/examples/api/integration_test/smoke_integration_test.dart [E]
  Failed to load "/home/bob/git/flutter/flutter/examples/api/integration_test/smoke_integration_test.dart": Unable to generate build files
  package:flutter_tools/src/base/common.dart 10:3                    throwToolExit
  package:flutter_tools/src/linux/build_linux.dart 174:5             _runCmake
  ===== asynchronous gap ===========================
  package:flutter_tools/src/linux/build_linux.dart 81:5              buildLinux
  ===== asynchronous gap ===========================
  package:flutter_tools/src/linux/linux_device.dart 67:5             LinuxDevice.buildForDevice
  ===== asynchronous gap ===========================
  package:flutter_tools/src/desktop_device.dart 120:7                DesktopDevice.startApp
  ===== asynchronous gap ===========================
  package:flutter_tools/src/test/integration_test_device.dart 55:39  IntegrationTestTestDevice.start
  ===== asynchronous gap ===========================
  package:flutter_tools/src/test/flutter_platform.dart 546:49        FlutterPlatform._startTest.<fn>
  ===== asynchronous gap ===========================
  package:flutter_tools/src/base/async_guard.dart 111:24             asyncGuard.<fn>
  
00:00 +0 -1: Some tests failed.
Running "/home/bob/git/flutter/flutter/bin/flutter test --reporter=expanded --device-id=linux /home/bob/git/flutter/flutter/examples/api/integration_test/smoke_integration_test.dart" in /home/bob/git/flutter/flutter/examples/api exited with code 1
```

The above is running `flutter test` from `examples/api`, the change in this PR changes to `examples/api/integration_test`.

I don't know why CMake is failing in the other directory and not enough about flutter test to be sure this change is valid. @stuartmorgan do you know if this change is suitable or can propose something else?

chunhtai commented:
(triage) I am not a Linux expert, but looks like there may be a directory mismatch? cc @stuartmorgan, can you take a look at this pr again?

stuartmorgan commented:
Sorry, I missed that question. I'm not sure what the implications on flutter/flutter CI of changing the working directory for the test would be; a repo expert would be better equipped to answer that.

I'm confused as to why we would need to though, because the code that is throwing passes an explicit working directory to `cmake`. @robert-ancell What are the values of `sourceDir` and `buildDir` [here](https://github.com/flutter/flutter/blob/b8dcb0c3c577cf476bf5bcc654b599724e13edde/packages/flutter_tools/lib/src/linux/build_linux.dart#L164-L166) with and without that change?

chunhtai commented:
(triage) Hi @robert-ancell , do you have plan to come back to this PR? Looks like Stuart had pointed some direction to triage the error.

goderbauer commented:
(triage): Spoke to @robert-ancell last week and he said this is back on his radar.

goderbauer commented:
(triage): I am going to close this for now since there hasn't been any follow-up. Feel free to reopen (or create a new PR) when you have the time to work on this again.
